src/System.Drawing.Common/src/System/Drawing/Drawing2D/GraphicsPath.Windows.cs

```diff
@@ -183,46 +183,36 @@ public FillMode FillMode
             }
         }
 
-        private PathData _GetPathData()
+        private unsafe PathData _GetPathData()
         {
-            int ptSize = Marshal.SizeOf(typeof(GPPOINTF));
+            int count = PointCount;
 
-            int numPts = PointCount;
+            PathData pathData = new PathData()
+            {
+                Types = new byte[count],
+                Points = new PointF[count]
+            };
 
-            PathData pathData = new PathData() { Types = new byte[numPts] };
+            if (count == 0)
+                return pathData;
 
-            IntPtr memoryPathData = Marshal.AllocHGlobal(3 * IntPtr.Size);
-            IntPtr memoryPoints = Marshal.AllocHGlobal(checked(ptSize * numPts));
-            try
+            fixed (byte* t = pathData.Types)
+            fixed (PointF* p = pathData.Points)
             {
-                GCHandle typesHandle = GCHandle.Alloc(pathData.Types, GCHandleType.Pinned);
-                try
+                GpPathData data = new GpPathData
                 {
-                    IntPtr typesPtr = typesHandle.AddrOfPinnedObject();
+                    Count = count,
+                    Points = p,
+                    Types = t
+                };
 
-                    Marshal.StructureToPtr(numPts, memoryPathData, false);
-                    Marshal.StructureToPtr(memoryPoints, (IntPtr)((long)memoryPathData + IntPtr.Size), false);
-                    Marshal.StructureToPtr(typesPtr, (IntPtr)((long)memoryPathData + 2 * IntPtr.Size), false);
-
-                    int status = SafeNativeMethods.Gdip.GdipGetPathData(new HandleRef(this, nativePath), memoryPathData);
-
-                    if (status != SafeNativeMethods.Gdip.Ok)
-                    {
-                        throw SafeNativeMethods.Gdip.StatusException(status);
-                    }
+                int status = SafeNativeMethods.Gdip.GdipGetPathData(new HandleRef(this, nativePath), &data);
 
-                    pathData.Points = SafeNativeMethods.Gdip.ConvertGPPOINTFArrayF(memoryPoints, numPts);
-                }
-                finally
+                if (status != SafeNativeMethods.Gdip.Ok)
                 {
-                    typesHandle.Free();
+                    throw SafeNativeMethods.Gdip.StatusException(status);
```

Comment:
There's a helper for this:
```C#
SafeNativeMethods.Gdip.CheckStatus(status);
```
https://github.com/dotnet/corefx/blob/b5b75ee8969f2c67689eafe7b1ba73d0bddda019/src/System.Drawing.Common/src/System/Drawing/Gdiplus.cs#L261-L267

Reply:
Yes, I plan to rationalize out the variety of helpers in a separate change.